packages/trace-viewer/src/ui/shared/dialog.tsx

```diff
@@ -96,12 +98,24 @@ export const Dialog: React.FC<React.PropsWithChildren<DialogProps>> = ({
     };
   }, []);
 
+  React.useLayoutEffect(() => {
+    if (!dialogRef.current)
+      return;
+
+    if (open) {
+      if (isModal)
+        dialogRef.current.showModal();
+      else
+        dialogRef.current.show();
+    } else {
+      dialogRef.current.close();
+    }
+  }, [open, isModal]);
+
   return (
-    open && (
-      <dialog ref={dialogRef} style={style} className={className} data-testid={dataTestId} open>
-        {children}
-      </dialog>
-    )
+    <dialog ref={dialogRef} style={style} className={className} data-testid={dataTestId} open={false}>
```

Comment:
Do we still need the `open` param here?
```suggestion
    <dialog ref={dialogRef} style={style} className={className} data-testid={dataTestId}>
```